### DISCUSSION TITLE
New checks

### DISCUSSION THREAD

**Discussion by zupo** [ParetoSecurity/pareto-mac#3] [Ideas]
What checks should we add to Pareto Security app?

**Comment by zupo** [comment]
Filevault, definitely. I.e. this:

<img width="630" alt="Screenshot 2021-07-27 at 08 45 35" src="https://user-images.githubusercontent.com/311580/127108347-1be66749-b71c-4c13-b680-addde6fee75e.png">

**Reply by zupo** [reply]
Done: https://paretosecurity.com/security-checks/filevault

**Comment by zupo** [comment]
Support Lastpass, besides 1Password.

**Reply by jimmhay** [reply]
i believe Dashlane is also popular

**Comment by zupo** [comment]
A few more ideas:
* TimeMachine backup in last 10 days
* TimeMachine disk is encrypted
* Password hint disabled
* Root user login disabled
* Disk less than 90% full

**Reply by dmurko** [reply]
> TimeMachine backup in last 10 days

Keep in mind, TimeMachine has its own reminder.

**Reply by zupo** [reply]
Only if you have "Back Up Automatically" ticked in Time Machine preferences. I disable it from time to time, for example, when I travel, so that I don't get annoyed by it.

**Reply by jimmhay** [reply]
Yeah, +1 for including TM - having all checks in one place is convenient, and also makes more sense if using it with a group of devices.

What's the security benefit of being less than 90% full?

**Reply by zupo** [reply]
> What's the security benefit of being less than 90% full?

TimeMachine can do granual snapshots on the local disk: https://support.apple.com/en-us/HT204015

`Time Machine stores snapshots only on disks that have plenty of free space`. I don't know what exactly "plenty of free space" means though.

Another one: software upgrades can get stuck if disk is completely full.

There are also noticeable performance benefits.

**Reply by zupo** [reply]
This article says 15% free disk is the bare minimum to avoid performance drawbacks: https://www.lifewire.com/how-much-free-drive-space-needed-2260187

**Comment by zupo** [comment]
A bunch of checks that I'd personally love to have: https://github.com/kristovatlas/osx-config-check

**Reply by zupo** [reply]
Similar: https://github.com/alichtman/stronghold

**Comment by zupo** [comment]
Can we also check battery health?

**Reply by dz0ny** [reply]
https://github.com/ParetoSecurity/pareto-mac/releases/tag/0.0.11a for testing

**Reply by jimmhay** [reply]
I'm failing this test, despite buying my laptop less than a month ago. FruitJuice is saying 5 of 1000 cycles, 101% of factory capacity

**Reply by dz0ny** [reply]
Yep, this is only working now in debug builds, there are entitlements that we need to research to properly support it.

**Comment by karantan** [comment]
- not used installed apps
- not used apps that start on startup (and run in the background)

**Comment by dmurko** [comment]
MacOS: up-to-date

App Store: outdated apps or check automatic updates.

**Comment by dmurko** [comment]
- Remote Login
- Remote Management

<img width="411" alt="Screenshot 2021-08-16 at 16 14 09" src="https://user-images.githubusercontent.com/1309671/129577999-72f03964-5499-412e-8a28-3186047ede7a.png">

**Reply by zupo** [reply]
Done:
* https://paretosecurity.com/security-checks/remote-login
* https://paretosecurity.com/security-checks/remote-management

**Comment by dmurko** [comment]
Software Update

<img width="347" alt="Screenshot 2021-08-16 at 16 29 07" src="https://user-images.githubusercontent.com/1309671/129580104-5471d57c-ac2a-4c8e-bcb2-698e22878421.png">

**Comment by dmurko** [comment]
Find My

<img width="420" alt="Screenshot 2021-08-16 at 16 33 50" src="https://user-images.githubusercontent.com/1309671/129580832-a74bad66-8622-45ef-9907-212187f16b27.png">

**Comment by dmurko** [comment]
Check that all installed browsers are up-to-date.

**Reply by zupo** [reply]
Done: https://paretosecurity.com/security-checks/software-updates

**Comment by zupo** [comment]
Just had a chat with @reillychase and came up with another idea: iterate through `~/.ssh` and verify that all keys have a passphrase set.

**Reply by zupo** [reply]
This is now done and available in the recent release. More info on https://paretosecurity.com/security-checks/ssh-keys-password.

**Comment by zupo** [comment]
Make sure AirDrop is set to Contacts Only.

Submitted by a user.

**Reply by zupo** [reply]
Refs https://secure.helpscout.net/conversation/1814480971/36493?folderId=4882163

**Comment by DotTheI** [comment]
Apple Passwords

**Reply by zupo** [reply]
Can you expand on what you mean?

**Reply by DotTheI** [reply]
Sure. I mean, even though Passwords app is installed by default, PS says 'Password manager is not installed'
